source/rust_verify/src/rust_to_vir_expr.rs

```diff
@@ -312,12 +319,27 @@ fn fn_call_to_vir<'tcx>(
     let is_add = f_name == "core::ops::arith::Add::add";
     let is_sub = f_name == "core::ops::arith::Sub::sub";
     let is_mul = f_name == "core::ops::arith::Mul::mul";
-    let is_spec = is_admit || is_requires || is_ensures || is_invariant || is_decreases;
+    let is_spec = is_admit
+        || is_requires
+        || is_ensures
+        || is_invariant
+        || is_decreases
+        || is_opens_invariants_none
+        || is_opens_invariants_any
+        || is_opens_invariants
+        || is_opens_invariants_except;
     let is_quant = is_forall || is_exists;
     let is_directive = is_hide || is_reveal || is_reveal_fuel;
     let is_cmp = is_equal || is_eq || is_ne || is_le || is_ge || is_lt || is_gt;
     let is_arith_binary = is_add || is_sub || is_mul;
 
+    if f_name == BUILTIN_INV_BEGIN || f_name == BUILTIN_INV_END {
```

Comment:
Is this relying on the fact that the macro uses these functions when imported at the root of the crate? If that's the case, that may be brittle.

Edit: it's not. This code path isn't reached when using it correctly because the calls are matched by the `invariant_block_to_vir` directly.

Reply:
i'll add a comment as explanation